src/python/WMCore/WMSpec/StdSpecs/StdBase.py

```diff
@@ -391,12 +391,14 @@ def setupProcessingTask(self, procTask, taskType, inputDataset=None, inputStep=N
         procTaskCmsswHelper.setOverrideCatalog(self.overrideCatalog)
         procTaskCmsswHelper.setErrorDestinationStep(stepName=procTaskLogArch.name())
 
-        if forceMerged:
+        if forceMerged and isinstance(forceMerged, bool):
             procTaskStageHelper.setMinMergeSize(0, 0)
-        elif forceUnmerged:
+        elif forceUnmerged and isinstance(forceUnmerged, bool):
             procTaskStageHelper.disableStraightToMerge()
         else:
             procTaskStageHelper.setMinMergeSize(self.minMergeSize, self.maxMergeEvents)
+            if forceUnmerged and isinstance(forceUnmerged, list):
```

Comment:
Two comments here: 
1) as my previous comments, I suggest keeping only `if forceUnmerged:`, which will check for empty list or list with elements.
2) this code won't work! IF it goes into the `elif` in L396, it will never go into the `else` below. IF it goes into the `else` in L398, it will never have a true forceUnmerged list.